codegenerator/cli/templates/static/codegen/src/Benchmark.res

```diff
@@ -72,27 +115,58 @@ module Data = {
     self.millisAccum->MillisAccum.increment(label, amount)
   }
 
-  let addSummaryData = (self: t, ~group, ~label, ~value) => {
-    self.summaryData->SummaryData.add(~group, ~label, ~value)
+  let addSummaryData = (self: t, ~group, ~label, ~value, ~decimalPlaces=2) => {
+    self.summaryData->SummaryData.add(~group, ~label, ~value, ~decimalPlaces)
+  }
+}
+
+module LazyWriter = {
+  let isWriting = ref(false)
+  let scheduledWriteFn: ref<option<unit => promise<unit>>> = ref(None)
+  let lastRunTimeMillis = ref(0.)
+
+  let rec start = async () => {
+    switch scheduledWriteFn.contents {
+    | Some(fn) =>
+      isWriting := true
+      scheduledWriteFn := None
+      lastRunTimeMillis := Js.Date.now()
+
+      switch await fn() {
+      | exception exn => Logging.errorWithExn(exn, "Failed to write benchmark cache file")
+      | _ => ()
+      }
+      isWriting := false
+      await start()
+    | None => ()
+    }
+  }
+
+  let schedule = (~intervalMillis=500, fn) => {
+    scheduledWriteFn := Some(fn)
+    if !isWriting.contents {
+      let timeSinceLastRun = Js.Date.now() -. lastRunTimeMillis.contents
+      if timeSinceLastRun >= intervalMillis->Belt.Int.toFloat {
+        start()->ignore
+      } else {
+        let _ = Js.Global.setTimeout(() => {
+          start()->ignore
+        }, intervalMillis - timeSinceLastRun->Belt.Float.toInt)
```

Comment:
This can be scheduled multiple times. Nothing should break, just a risk of unexpected execution.

Reply:
It can but it overwrites the scheduled function on each schedule. So more like it just calls start multiple times.

Maybe it's best to just add the "isWriting" if check here or in the start function again.

Reply:
Added the check in "start" 👍🏼